docs/proposals/20201024-advancedcronjob.md

```diff
@@ -0,0 +1,95 @@
+---
+title: AdvancedCronJob Crd and Controller
+authors:
+  - "@rishi-anand"
+reviewers:
+  - "@Fei-Guo"
+  - "@FillZpp"
+  - "@jzhoucliqr"
+creation-date: 2020-10-25
+last-updated: 2020-10-25
+status: implementable
+---
+
+# Implementing AdvancedCronJob Crd and Controller
+- Implementing AdvancedCronJob to support Job/BroadcastJob or any other future CRD and to run it periodically at a given schedule. 
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a proposal and for highlighting
+any additional information provided beyond the standard proposal template.
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc) a table of contents from markdown are available.
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+  - [Implementation History](#implementation-history)
+
+## Summary
+
+This controller will be very generic and will have implementations to help developers to run a job or any CRD in specific schedule.
+
+## Motivation
+
+- Developer may come across a use-case when some job needs to be executed on at a specific schedule.
+- Found same use-case requirement in Issues #251 and I got motivated to implement it
+
+### Goals
+
+- Implementing a custom controller for AdvancedCronJob which acts like CronJob but it schedules Job/BroadcastJob or other CRD
+
+## Proposal
+
+- Adding a new CRD and controller for AdvancedCronJob.
+- AdvancedCronJob should be able to reconcile Job/BroadcastJob or any other future CRD if required.
+- Once AdvancedCronJob is created, spec cannot be modified.
+- Adding webhook for validation of AdvancedCronJob.
+
+### User Stories
+
+- Implement a CRD which contains below fields and a controller which honors all the fields and reconciles accordingly.
+```
```

Comment:
For API definition, can you add struct definition instead of CR yaml file?

I would suggest to introduce a JobTemplate struct that contains all the supported job types:
AdvancedCronJobSpec struct{
     template JobTemplate
}
type JobTemplate struct {
     JobTemplate *batchv1beta1.JobTemplateSpec
     BroadcastJobTemplate *BroadcastJobTemplateSpec
}